wp-content/civi-extensions/goonjcustom/Civi/GoonjActivitiesService.php

```diff
@@ -0,0 +1,539 @@
+      ->setLimit(1)
+      ->execute()->single();
+
+    $inductionId = $induction['id'];
+
+    EckEntity::update('Collection_Camp', FALSE)
+      ->addValue('Collection_Camp_Intent_Details.Initiator_Induction_Id', $inductionId)
+      ->addWhere('id', '=', $collectionCampId)
+      ->execute();
+  }
+
+  /**
+   * This hook is called after the database write on a custom table.
+   *
+   * @param string $op
+   *   The type of operation being performed.
+   * @param string $objectName
+   *   The custom group ID.
+   * @param int $objectId
+   *   The entityID of the row in the custom table.
+   * @param object $objectRef
+   *   The parameters that were sent into the calling function.
+   */
+  public static function setOfficeDetails($op, $groupID, $entityID, &$params) {
+    if ($op !== 'create' || self::getEntitySubtypeName($entityID) !== self::ENTITY_SUBTYPE_NAME) {
+      return;
+    }
+
+    if (!($stateField = self::findStateField($params))) {
+      return;
+    }
+
+    $stateId = $stateField['value'];
+    $goonjActivitiesId = $stateField['entity_id'];
+
+    if (!$stateId) {
+      \CRM_Core_Error::debug_log_message('Cannot assign Goonj Office to goonj activities: ' . $goonjActivitiesId);
+      \CRM_Core_Error::debug_log_message('No state provided on the intent for goonj activities: ' . $goonjActivitiesId);
+      return FALSE;
+    }
+
+    $officesFound = Contact::get(FALSE)
+      ->addSelect('id')
+      ->addWhere('contact_type', '=', 'Organization')
+      ->addWhere('contact_sub_type', 'CONTAINS', 'Goonj_Office')
+      ->addWhere('Goonj_Office_Details.Collection_Camp_Catchment', 'CONTAINS', $stateId)
+      ->execute();
+
+    $stateOffice = $officesFound->first();
+
+    // If no state office is found, assign the fallback state office.
+    if (!$stateOffice) {
+      $stateOffice = self::getFallbackOffice();
+    }
+
+    $stateOfficeId = $stateOffice['id'];
+
+    $rs = EckEntity::update('Collection_Camp', FALSE)
+      ->addValue('Goonj_Activities.Goonj_Office', $stateOfficeId)
+      ->addWhere('id', '=', $goonjActivitiesId)
+      ->execute();
+
+    $coordinators = Relationship::get(FALSE)
+      ->addWhere('contact_id_b', '=', $stateOfficeId)
+      ->addWhere('relationship_type_id:name', '=', self::RELATIONSHIP_TYPE_NAME)
+      ->execute();
+    $RELATIONSHIP_TYPE_NAME = self::RELATIONSHIP_TYPE_NAME;
+
+    $coordinatorCount = $coordinators->count();
+
+    if ($coordinatorCount === 0) {
+      $coordinator = self::getFallbackCoordinator();
+    }
+    elseif ($coordinatorCount > 1) {
+      $randomIndex = rand(0, $coordinatorCount - 1);
+      $coordinator = $coordinators->itemAt($randomIndex);
+    }
+    else {
+      $coordinator = $coordinators->first();
+    }
+
+    if (!$coordinator) {
+      \CRM_Core_Error::debug_log_message('No coordinator available to assign.');
+      return FALSE;
+    }
+
+    $coordinatorId = $coordinator['contact_id_a'];
+
+    $rs1 = EckEntity::update('Collection_Camp', FALSE)
```

Comment:
Please use better variable naming and i don't think we need variable here as it's an update function